File: app/src/main/java/com/pnuema/android/bible/statics/DeepLinks.java
package com.pnuema.android.bible.statics;

import android.content.Intent;
import android.net.Uri;
import android.text.TextUtils;

public final class DeepLinks {
    public static void handleDeepLinks(Intent intent) {
        Uri data = intent.getData();

        if (data == null
                || (TextUtils.isEmpty(data.getQueryParameter("book")) && TextUtils.isEmpty(data.getQueryParameter("bookid")))
                || TextUtils.isEmpty(data.getQueryParameter("chapter"))
                || TextUtils.isEmpty(data.getQueryParameter("verse"))) {
            return;
        }

        String bookAbb = data.getQueryParameter("book");
        //TODO look up book if it was provided to get the book id
        Integer bookid = Integer.valueOf(data.getQueryParameter("bookid"));
        Integer chapter = Integer.valueOf(data.getQueryParameter("chapter"));
        Integer verse = Integer.valueOf(data.getQueryParameter("verse"));

        CurrentSelected.setBook(bookid);
        CurrentSelected.setChapter(chapter);
        CurrentSelected.setVerse(verse);
        CurrentSelected.savePreferences();
    }
}

File: app/src/main/java/com/pnuema/bible/android/statics/DeepLinks.kt
package com.pnuema.bible.android.statics

import android.content.Intent
import android.text.TextUtils
import kotlinx.coroutines.runBlocking
import java.util.*

object DeepLinks {
    fun handleDeepLinks(intent: Intent) {
        val data = intent.data ?: return

        if (TextUtils.isEmpty(data.getQueryParameter("book")) && TextUtils.isEmpty(data.getQueryParameter("bookid"))
                || TextUtils.isEmpty(data.getQueryParameter("chapter"))
                || TextUtils.isEmpty(data.getQueryParameter("verse"))) {
            return
        }

        //val bookAbb = data.getQueryParameter("book")
        //TODO look up book if it was provided to get the book id
        val bookid = Integer.valueOf(Objects.requireNonNull<String>(data.getQueryParameter("bookid")))
        val chapter = Integer.valueOf(Objects.requireNonNull<String>(data.getQueryParameter("chapter")))
        val verse = Integer.valueOf(Objects.requireNonNull<String>(data.getQueryParameter("verse")))

        CurrentSelected.book = bookid
        CurrentSelected.chapter = chapter
        CurrentSelected.verse = verse

        runBlocking {
            CurrentSelected.savePreferences()
        }
    }
}
